### PR TITLE
Medianet: Add fledge support

### PR DESCRIPTION
**Pull request by ghost**
Add Fledge support for Medianet adapter. The response from bidder for fledge auction config support the ortb format mentioned in the below link.
https://github.com/InteractiveAdvertisingBureau/openrtb/blob/main/extensions/community_extensions/Protected%20Audience%20Support.md

### CODE REVIEW

File: adapters/medianet/medianet.go [hunk 1]
```diff
@@ -79,14 +101,18 @@ func (a *adapter) MakeBids(internalRequest *openrtb2.BidRequest, externalRequest
...
+
 	return bidResponse, errs
 }
 
 // Builder builds a new instance of the Medianet adapter for the given bidder with the given config.
 func Builder(bidderName openrtb_ext.BidderName, config config.Adapter, server config.Server) (adapters.Bidder, error) {
 	url := buildEndpoint(config.Endpoint, config.ExtraAdapterInfo)
-	return &adapter{
-		endpoint: url,
+	return &MedianetAdapter{
```
[the remark this answers is not in the document]
Comment (b-chinmay) [reply adapters/medianet/medianet.go]: Done.

File: adapters/medianet/medianet.go [hunk 2]
```diff
@@ -106,6 +132,31 @@ func getMediaTypeForImp(impID string, imps []openrtb2.Imp) (openrtb_ext.BidType,
 	}
 }
 
+func extractFledge(a *adapter, bidResp openrtb2.BidResponse) []*openrtb_ext.FledgeAuctionConfig {
+	var fledgeAuctionConfigs []*openrtb_ext.FledgeAuctionConfig
+
+	var bidRespExt medianetRespExt
+	if err := json.Unmarshal(bidResp.Ext, &bidRespExt); err != nil {
+		return nil
+	}
```
Comment (onkarvhanumante) [inline adapters/medianet/medianet.go:141]: any particular reason to not return error from line 140

Comment (b-chinmay) [reply adapters/medianet/medianet.go]: Done, took error into consideration for assigning fledge config.

**Comment by onkarvhanumante** [comment]
PR tests are failing. 

```
k  	github.com/prebid/prebid-server/v2/adapters/marsmedia	0.026s	coverage: 74.3% of statements
--- FAIL: TestJsonSamples (0.00s)
    test_json.go:130: medianettest/exemplary/fledge.json: MakeBids had wrong error count. Expected 0, got 1 ([Unable to fetch mediaType for imp: 1])
FAIL
```

**Comment by onkarvhanumante** [comment]
PR author account has been deleted. Therefore closing this account

image

**Pull request closed without merging**